### PR TITLE
docs: Consul K8s Overview update

### PR DESCRIPTION
**Pull request by david-yu**
- Remove deploy client and servers as use cases as they are rather deployment models.
- Change link for getting started to point to the Getting Started for K8s Service Mesh guide 
- Separate out Architecture into its own separate page

### CODE REVIEW

File: website/content/docs/k8s/architecture.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/architecture.mdx:20]:
```suggestion
This topic describes the architecture, components, and resources associated with Consul deployments to Kubernetes. Consul employs the same architectural design on Kubernetes as it does with other platforms (see [Architecture](/docs/architecture)), but Kubernetes provides additional benefits that make operating a Consul cluster easier. 

Refer to the standard [production deployment guide](https://learn.hashicorp.com/consul/datacenter-deploy/deployment-guide) for important information, regardless of the deployment platform.
```
I just want to be clear about what this topic covers.

File: website/content/docs/k8s/architecture.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/architecture.mdx:24]:
```suggestion
The server agents are run as a **StatefulSet**, using persistent volume
```
I think that because the Kubernetes keywords are configuration settings (or appear to be based on my Google search), we should use code font and reposition how we reference them to make this clear. 

For example, we might want to rephrase this sentence to something like:

Consul server agents operate as `StatefulSets` types and use persistent . . . "

More precise, no?

File: website/content/docs/k8s/architecture.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/architecture.mdx:37]:
```suggestion
Configure and register a Kubernetes `Service` to represent the servers and expose the various
ports. The service's DNS address joins the servers to each other without requiring access to the Kubernetes cluster. You should configure the service to publish endpoints that may not be ready so that the endpoints can join servers during bootstrap and upgrades.
```
I think that one of the confusing bits (for me, anyway) is that this is written in passive voice. It sounds as if this all happens automatically when you install Consul. Even if the operations are performed programmatically without a clear "you", we should try to use active/direct language so it's clear that we are directing users to do something.

Also, I think this suggestion retains the meaning, but I was a little unclear about that last sentence.

File: website/content/docs/k8s/architecture.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/architecture.mdx:39]: See previous comment about passive voice and code font for K8s entities.

**Review by trujillo-adam — APPROVED** [review]
A few suggestions and question; nothing that has to hold up this change.

File: website/content/docs/k8s/architecture.mdx
Comment (david-yu) [reply website/content/docs/k8s/architecture.mdx]: Thanks!

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/608852.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 55e864d1257dd1a55e517911fea16772448728be onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 55e864d1257dd1a55e517911fea16772448728be onto `release/1.11.x` succeeded!

**Pull request merged**
